app/controllers/my-projects/project/hearing/add.js

```diff
@@ -0,0 +1,80 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import moment from 'moment';
+
+// create a hearing record when a user clicks on the submit hearing button
+function createHearing(thisHearing, hearingLocation, hearingDate) {
+  const newHearing = thisHearing.store.createRecord('hearing', {
+    project: thisHearing.model,
+    location: hearingLocation,
+    date: hearingDate,
+  });
+  newHearing.save();
+}
+
+export default class MyProjectsProjectHearingAddController extends Controller {
+  @service
+  store;
+
+  // for checking if user clicks submit button without inputting necessary attributes
+  dateMissing;
+
+  timeMissing;
+
+  locationMissing;
+
+  // options for ember-timepicker
+  timeOptions= {
+    dropdown: true,
+    interval: 5,
+    scrollbar: true,
+    minTime: '6:00 AM',
+    maxTime: '11:00 PM',
+    startTime: '6:00',
+  };
+
+  // minimum date for calendar
+  currentDate = new Date();
+
+  // for clearing search results on the geosearch input
+  onClearSearchResult = () => {};
+
+  // when a user clicks on the x in the input box
+  @action
+  clearSearchResult() {
+    this.onClearSearchResult();
+  }
+
+  // action triggered when click on "submit hearing" button
+  @action
+  saveHearingForm() {
+    // arguments to trigger a notification to the user that they need to input a field
+    if (!this.date) { this.set('dateMissing', true); } else { this.set('dateMissing', false); }
+    if (!this.location) { this.set('locationMissing', true); } else { this.set('locationMissing', false); }
+    if (!this.time) { this.set('timeMissing', true); } else { this.set('timeMissing', false); }
```

Comment:
Another way this can be done is `this.set('dateMissing', !!this.date);`

Reply:
cool cool, will edit